src/sql/azdata.proposed.d.ts

```diff
@@ -1098,6 +1098,10 @@ declare module 'azdata' {
 			 * Extension can store additional information that the provider needs to uniquely identify a table.
 			 */
 			[key: string]: any;
+			/**
+			 * Gives the table type.
+			 */
+			type?: TableType;
```

Comment:
this is a mssql specific thing, please remove it from the typing and add it to mssql extension.